src/main/java/org/isf/accounting/gui/BillBrowser.java

```diff
@@ -1231,58 +1201,28 @@ private void updateTotals() {
 		}
 		
 		totalPeriod = new BigDecimal(0);
-		balancePeriod = new BigDecimal(0);
+
 		totalToday = new BigDecimal(0);
 		balanceToday = new BigDecimal(0);
 		userToday = new BigDecimal(0);
 		userPeriod = new BigDecimal(0);
 		
-		ArrayList<Integer> notDeletedBills = new ArrayList<Integer>();
-				
+		List<Integer> notDeletedBills = billPeriod.stream()
+				.filter(bill -> !bill.getStatus().equals("D"))
+				.map(Bill::getId)
+				.collect(Collectors.toList());
+
 		//Bills in range contribute for Not Paid (balance)
-		for (Bill bill : billPeriod) {
-			if (!bill.getStatus().equals("D")) {
-				notDeletedBills.add(bill.getId());
-				BigDecimal balance = new BigDecimal(Double.toString(bill.getBalance()));
-				balancePeriod = balancePeriod.add(balance);
-			}
-		}
-		
-		//Payments in range contribute for Paid Period (total)
-		for (BillPayments payment : paymentsPeriod) {
-			if (notDeletedBills.contains(payment.getBill().getId())) {
-				BigDecimal payAmount = new BigDecimal(Double.toString(payment.getAmount()));
-				String payUser = payment.getUser();
-				
-				totalPeriod = totalPeriod.add(payAmount);
-					
-				if (!GeneralData.SINGLEUSER && payUser.equals(user))
-					userPeriod = userPeriod.add(payAmount);
-			}
-		}
-		
+		balancePeriod = new BalanceTotal(billPeriod).getValue();
```

Comment:
So regarding to UserTotal instead of method is just coding style (I prefer to create ValueObject instead of another helper method). Anyway I agree that it should be in core in next step. 

https://martinfowler.com/bliki/ValueObject.html - pattern description

Reply:
Thank you for the explanation. I can see the idea now, but I would not call it Value Object pattern, which aim seems just to hold data and to guarantee immutabily and comparation. Those classes look more like isolated methods, where you set values and ask for the result (more similar to [Strategy Pattern](https://www.tutorialspoint.com/design_pattern/strategy_pattern.htm)).

Anyway very good, we like the idea to isolate those chunks into tested code, so we can move them in managers in a second step 👍.